sphinx/ext/autodoc/__init__.py

```diff
@@ -396,9 +396,9 @@ def format_signature(self, **kwargs: Any) -> str:
                 except TypeError:
                     # retry without arguments for old documenters
                     args = self.format_args()
-            except Exception as err:
-                logger.warning(__('error while formatting arguments for %s: %s') %
-                               (self.fullname, err), type='autodoc')
+            except Exception:
+                logger.warning(__('error while formatting arguments for %s:') %
+                               self.fullname, type='autodoc', exc_info=True)
```

Comment:
Note: I thought showing full stacktrace is a bit noisy for users at first. On the other hand, it is very important for developers including extension's authors. In addition, `format_args()` is one of the handlers that is unlikely to throw an exception. So +1 for this change.

Reply:
It might be reasonable to respect a verbose flag here, as some future work.

Reply:
Absolutely!

Reply:
I made a patch for this purpose #7608